src/sentry/api/endpoints/organization_traces.py

```diff
@@ -930,22 +937,29 @@ def get_trace_name(trace):
             row["trace"]: row["count()"] for row in traces_occurrences_results["data"]
         }
 
-        return [
-            {
+        results: list[TraceResult] = []
+
+        for row in traces_metas_results["data"]:
+            info = get_trace_info(row["trace"])
+
+            result: TraceResult = {
                 "trace": row["trace"],
                 "numErrors": traces_errors.get(row["trace"], 0),
                 "numOccurrences": traces_occurrences.get(row["trace"], 0),
                 "matchingSpans": row[MATCHING_COUNT_ALIAS],
                 "numSpans": row["count()"],
-                "project": get_trace_name(row["trace"])[0],
-                "name": get_trace_name(row["trace"])[1],
+                "project": info[0],
+                "name": info[1],
+                "rootDuration": info[2],
                 "duration": row["last_seen()"] - row["first_seen()"],
                 "start": row["first_seen()"],
                 "end": row["last_seen()"],
                 "breakdowns": traces_breakdowns[row["trace"]],
             }
-            for row in traces_metas_results["data"]
-        ]
+
+            results.append(result)
```

Comment:
lol is this a separate line just because of typing?

Reply:
didnt wanna call `get_trace_name` three times here and typing was weird here for some reason